apps/api/v2/src/ee/bookings/2024-08-13/services/input.service.ts

```diff
@@ -0,0 +1,418 @@
+    const eventType = await this.eventTypesRepository.getEventTypeByIdWithOwnerAndTeam(
+      inputBooking.eventTypeId
+    );
+
+    if (!eventType) {
+      throw new NotFoundException(`Event type with id=${inputBooking.eventTypeId} not found`);
+    }
+
+    const startTime = DateTime.fromISO(inputBooking.start, { zone: "utc" }).setZone(
+      inputBooking.attendee.timeZone
+    );
+    const endTime = startTime.plus({ minutes: eventType.length });
+
+    return {
+      start: startTime.toISO(),
+      end: endTime.toISO(),
+      eventTypeId: inputBooking.eventTypeId,
+      timeZone: inputBooking.attendee.timeZone,
+      language: inputBooking.attendee.language || "en",
+      // todo(Lauris): expose after refactoring metadata https://app.campsite.co/cal/posts/zysq8w9rwm9c
+      // metadata: inputBooking.metadata || {},
+      metadata: {},
+      hasHashedBookingLink: false,
+      guests: inputBooking.guests,
+      // note(Lauris): responses with name and email are required by the handleNewBooking
+      responses: inputBooking.bookingFieldsResponses
+        ? {
+            ...inputBooking.bookingFieldsResponses,
+            name: inputBooking.attendee.name,
+            email: inputBooking.attendee.email,
+          }
+        : { name: inputBooking.attendee.name, email: inputBooking.attendee.email },
+    };
+  }
+
+  async createRecurringBookingRequest(
+    request: Request,
+    body: CreateRecurringBookingInput_2024_08_13
+  ): Promise<BookingRequest> {
+    // note(Lauris): update to this.transformInputCreate when rescheduling is implemented
+    const bodyTransformed = await this.transformInputCreateRecurringBooking(body);
+    const oAuthClientId = request.get(X_CAL_CLIENT_ID);
+
+    const newRequest = { ...request };
+    const userId = (await this.createBookingRequestOwnerId(request)) ?? undefined;
+    const oAuthParams = oAuthClientId
+      ? await this.createBookingRequestOAuthClientParams(oAuthClientId)
+      : DEFAULT_PLATFORM_PARAMS;
+
+    const location = request.body.meetingUrl;
+    Object.assign(newRequest, {
+      userId,
+      ...oAuthParams,
+      platformBookingLocation: location,
+      noEmail: !oAuthParams.arePlatformEmailsEnabled,
+    });
+
+    newRequest.body = bodyTransformed.map((event) => ({
+      ...event,
+    }));
+
+    return newRequest as unknown as BookingRequest;
+  }
+
+  async transformInputCreateRecurringBooking(inputBooking: CreateRecurringBookingInput_2024_08_13) {
+    const eventType = await this.eventTypesRepository.getEventTypeByIdWithOwnerAndTeam(
+      inputBooking.eventTypeId
+    );
+    if (!eventType) {
+      throw new NotFoundException(`Event type with id=${inputBooking.eventTypeId} not found`);
+    }
+    if (!eventType.recurringEvent) {
+      throw new NotFoundException(`Event type with id=${inputBooking.eventTypeId} is not a recurring event`);
+    }
+
+    const occurrance = recurringEventSchema.parse(eventType.recurringEvent);
```

Comment:
occurrence

Reply:
Fixed spelling in [refactor: spelling](https://github.com/calcom/cal.com/pull/16200/commits/f20d834b80ab7cd23f25e42d0dfb1ca23d70f9d6)